monitor_test.go

```diff
@@ -86,6 +91,29 @@ func verifyRsapssSignature(b64Sig, b64Key string) error {
 	return rsapss.VerifySignatureFromB64(digest, b64Sig, b64Key)
 }
 
+func verifyContentSignaturePki(x5u, signature, input string) error {
+	// GetX5U will retrieve the chain from the local file:// and verify
+	// it so we don't have to. the first cert in the returned slice is
+	// the end entity we need to verify the signature with.
+	certs, err := contentsignaturepki.GetX5U(x5u)
+	if err != nil {
+		return err
+	}
+	// Get the public key from the end-entity
+	eePubKey := certs[0].PublicKey.(*ecdsa.PublicKey)
+	// parse the json signature
+	sig, err := contentsignaturepki.Unmarshal(signature)
+	if err != nil {
+		return err
+	}
+	// make a templated hash
+	_, inputHash := contentsignaturepki.MakeTemplatedHash([]byte(input), sig.Mode)
+	if !ecdsa.Verify(eePubKey, inputHash, sig.R, sig.S) {
+		return fmt.Errorf("ecdsa signature verification failed")
+	}
```

Comment:
this fails with
```
--- FAIL: TestMonitorPass (2.81s)
    monitor_test.go:82: {Ref:1t9ngg0sz77go3t1rvm8ms4ztg Type:contentsignaturepki Mode:p384ecdsa SignerID:remote-settings PublicKey:-----BEGIN CERTIFICATE-----
        MIICWzCCAeKgAwIBAgIIFY4HHiYeU5AwCgYIKoZIzj0EAwMwXzELMAkGA1UEBhMC
        VVMxCzAJBgNVBAgTAkNBMRYwFAYDVQQHEw1Nb3VudGFpbiBWaWV3MRAwDgYDVQQK
        EwdNb3ppbGxhMRkwFwYDVQQDExBjc3Jvb3QxNTUzMTg2NzQ3MB4XDTE5MDEyMDE3
        NDU0N1oXDTI5MDMyMTE2NDU0N1owYDELMAkGA1UEBhMCVVMxCzAJBgNVBAgTAkNB
        MRYwFAYDVQQHEw1Nb3VudGFpbiBWaWV3MRAwDgYDVQQKEwdNb3ppbGxhMRowGAYD
        VQQDExFjc2ludGVyMTU1MzE4Njc0NzB2MBAGByqGSM49AgEGBSuBBAAiA2IABE/l
        1sHSqAN6/ghHNWS4cNh/ZMNzxiSk0VadD8qic3URaswhY0B6444mCOzDCrT4zYnT
        9M1ydiW2zm1VmHowVgrBgcb82WRNlnnnBvcYrLwmSgUf9fvhObejTqF/QG4J9qNq
        MGgwDgYDVR0PAQH/BAQDAgGGMBMGA1UdJQQMMAoGCCsGAQUFBwMDMA8GA1UdEwEB
        /wQFMAMBAf8wMAYDVR0eAQH/BCYwJKAiMCCCHi5jb250ZW50LXNpZ25hdHVyZS5t
        b3ppbGxhLm9yZzAKBggqhkjOPQQDAwNnADBkAjBPi4uhzbCzoFxYD2LZY+hUQ1kX
        eHE9YmF0NOBbosIY041UrbLQLpQRXi4LFku9008CMGoSf/UGB6qV9KzsCY3s51Lp
        Rw/oK58D2n63FpPnOTICWvWAYsG3HPNCoURfLqlEOQ==
        -----END CERTIFICATE-----
         Signature:4gvaqXhERK8I2JnfJsaJBSCOuAckGKCfpW3n5shn9q3noOYqXWmodxVHzgPBtyW7QHvCGggTHoVskZxDJFFUZKd7U4Ppf1UR2rYEdnkCsGZ9z0ET1fRK_hC5LrRipS6P SignedFile: X5U:file:///tmp/autograph/chains/remotesettingsdev/remote-settings.content-signature.mozilla.org-20190421.chain}
    monitor_test.go:83: verification of monitoring response 4 failed: ecdsa signature verification failed
```

but I'm failing to see why. 302 ☕️

Reply:
I looked at this code in 10 different ways, printed all the intermediate values, checked the hash, and I still can't find why it fails... @g-k: would you mind having a look to see if I missed something here?

Reply:
FOUND IT! `certs[0]` is the intermediate, not the end-entity... 😫